src/Executor/Executor.php

```diff
@@ -173,10 +174,16 @@ public static function defaultFieldResolver($objectValue, $args, $contextValue,
         $fieldName = $info->fieldName;
         $property  = null;
 
-        if (is_array($objectValue) || $objectValue instanceof ArrayAccess) {
+        if (is_array($objectValue)) {
             if (isset($objectValue[$fieldName])) {
                 $property = $objectValue[$fieldName];
             }
+        } elseif ($objectValue instanceof ArrayAccess) {
```

Comment:
I think we should add a few lines of comment here to explain the rationale behind why this is treated differently from plain arrays.

Reply:
Extra docs added.

Reply:
Nicely done, thank you